File: src/1401-1500/1431 - Kids With the Greatest Number of Candies/KidsWithTheGreatestNumberOfCandies.java
import java.util.ArrayList;
import java.util.List;

class Solution {
  public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
    int max = 0;

    for (int candy : candies) {
      if (candy > max) {
        max = candy;
      }
    }

    List<Boolean> res = new ArrayList<>();

    for (int i = 0; i < candies.length; i++) {
      if (candies[i] + extraCandies >= max) {
        res.add(true);
      } else {
        res.add(false);
      }
    }

    return res;
  }
}

File: src/1401-1500/1431 - Kids With the Greatest Number of Candies/kids_with_the_greatest_number_of_candies.rs
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let mut max = 0;

        for i in 0..candies.len() {
            if candies[i] > max {
                max = candies[i];
            }
        }

        let mut result = Vec::new();

        for i in 0..candies.len() {
            if candies[i] + extra_candies >= max {
                result.push(true);
            } else {
                result.push(false);
            }
        }

        result
    }
}
